### DISCUSSION TITLE
No cross-platform lock: does it take python version markers into account?

### DISCUSSION THREAD

**Discussion by pawamoy** [pdm-project/pdm#1909] [Q&A] [answered]
I just locked dependencies with `--no-cross-platform`, and upon trying to install with Python 3.7 I get the following error:

```console
% pdm use -f python3.7 && pdm install
Using the last selection, add '-i' to ignore it.
Using Python interpreter: /home/pawamoy/.basher-packages/pyenv/pyenv/shims/python3.7 (3.7)
Using __pypackages__ because non-venv Python is used.
Updating executable scripts...
See /tmp/pdm-install-resolve-9700ffmb.log for detailed debug log.
[ResolutionImpossible]: [RequirementInformation(requirement=NamedRequirement(name='importlib-metadata', marker=<Marker('python_version < "3.8"')>, extras=set(), 
specifier=<SpecifierSet('')>, editable=False, prerelease=False), parent=None)]
Add '-v' to see the detailed traceback
```

I re-locked without the flag, and then was able to install dependencies with Python 3.7.

I thought `--no-cross-platform` would only ignore platform markers, not python version ones.

**Comment by frostming** [comment, marked as answer]
> I thought `--no-cross-platform` would only ignore platform markers, not python version ones.

It also considers the Python version, in fact the wheels and dependencies will be evaluated against the current Python interpreter(compared to not evaluated at all when the flag isn't given).

**Reply by pawamoy** [reply]
Ok, thanks. I'll use it CI only then!
